Martinh/faq

martin0995 opened the pull request:
### Description

Created FAQs page under the Learning section based on most common Discord questions.

### Checklist

- [x] **Required** - I have added a label to this PR 🏷️
- [x] **Required** - I have run my changes through Grammarly
- [ ] If pages have been moved, I have created redirects in the `wormhole-mkdocs` repo

eshaben commented on learn/faq/index.md:
````diff
@@ -0,0 +1,194 @@
+???- example "Example `overrides.json`"
+    ```json
+    {
+    "chains": {
+            "Bsc": {
+            "rpc": "http://127.0.0.1:8545"
+            },
+            "Sepolia": {
+            "rpc": "http://127.0.0.1:8546"
+            },
+            "Solana": {
+            "rpc": "http://127.0.0.1:8899"
+            }
+        }
+    }
````
Formatting is off here. Indented lines should be 4 spaces

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+Yes, we have an example of cross-chain lending that leverages [Wormhole’s Token Bridge](/learn/messaging/token-nft-bridge/){target=\_blank}. In this example, collateral deposits (such as ETH on Ethereum) are bridged to a hub chain. Once the collateral is deposited, the borrowed assets, like wrapped BNB, are bridged to Binance Smart Chain. You can explore the full implementation in this [cross-cha ...
+
+Alternatively, you can also implement cross-chain lending using [Wormhole’s core messaging](/learn/messaging/native-token-transfers/){target=\_blank} instead of the token bridge, which avoids the limitations imposed by governor limits. ETH would be custodied on Ethereum, and BNB on the Binance spoke during this setup. When a user deposits ETH on Ethereum, a core bridge message is sent to the hub  ...
```
```suggestion
Alternatively, you can also implement cross-chain lending using [Wormhole’s core messaging](/learn/messaging/native-token-transfers/){target=\_blank} instead of the Token Bridge, which avoids the limitations imposed by governor limits. ETH would be custodied on Ethereum, and BNB on the Binance spoke during this setup. When a user deposits ETH on Ethereum, a core bridge message is sent to the hub for accounting purposes. The hub then emits a message that can be redeemed on Binance to release the BNB. This approach allows for more direct asset control across chains while reducing reliance on Token Bridge limits.
```

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+This error typically occurs when the [Wormhole SDK](https://github.com/wormhole-foundation/wormhole-sdk-ts){target=\_blank} cannot recognize or register the necessary EVM protocols, which are required for interacting with Ethereum-based networks. The most common reason for this error is that the relevant EVM package for Wormhole's NTT has not been imported correctly.
+
+To resolve this issue, ensure you have imported the appropriate Wormhole SDK package for EVM environments. The necessary package for handling Native Token Transfers (NTT) on EVM chains is `@wormhole-foundation/sdk-evm-ntt`. Here's the correct import statement:
```
If you're going to spell out NTT, you only need to do it once when you first mention it, so it should be the first paragraph instead of right here

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+### Does the NTT manager function as an escrow account for a hubchain?
+
+Yes, the NTT manager acts like an escrow account for non-transferable tokens on a hubchain. To manage non-transferable tokens, you would add the NTT manager to the allowlist, ensuring that only the NTT manager can hold and control the tokens as they are transferred across chains.
```
```suggestion
Yes, the NTT manager acts like an escrow account for non-transferable tokens on a hub chain. To manage non-transferable tokens, you would add the NTT manager to the allowlist, ensuring that only the NTT manager can hold and control the tokens as they are transferred across chains.
```

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+### Which functions or events does Connect rely on for NTT integration?
+
+Connect relies on the NTT SDK for integration, with platform-specific implementations for both [Solana](https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/solana/ts/sdk/ntt.ts){target=\_blank} and [EVM](https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/evm/ts/src/ntt.ts){target=/_blank}. The key methods involved include:
```
```suggestion
Connect relies on the NTT SDK for integration, with platform-specific implementations for both [Solana](https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/solana/ts/sdk/ntt.ts){target=\_blank} and [EVM](https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/evm/ts/src/ntt.ts){target=\_blank}. The key methods involved include:
```

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+To manually submit a VAA (Verifiable Action Approval) to a destination chain, follow these steps:
+
+1. **Obtain the VAA in Base64 format** - navigate to the Advanced tab in [WormholeScan](https://wormholescan.io/){target=\_blank} to find the VAA associated with the transaction you want to submit and copy the VAA in base64 format
```
```suggestion
1. **Obtain the VAA in Base64 format** - navigate to the **Advanced** tab in [Wormholescan](https://wormholescan.io/){target=\_blank} to find the VAA associated with the transaction you want to submit and copy the VAA in base64 format
```

eshaben commented on learn/faq/index.md:
```diff
@@ -0,0 +1,194 @@
+
+Previously referred to as specialized relayers, custom relayers allow you to build and tailor relayers to fit your specific use case. To get started, refer to the following resources:
+
+ - [Custom Relayers documentation](https://wormhole.com/docs/learn/infrastructure/relayer/#custom-relayers){target=\_blank}
+ - [Relayer Engine](https://wormhole.com/docs/infrastructure/relayers/run-relayer/#get-started-with-the-relayer-engine){target=\_blank}
+ - [Run a Custom Relayer](https://wormhole.com/docs/infrastructure/relayers/run-relayer/){target=\_blank}
```
remove wormhole.com from the links please

eshaben reviewed (commented):
Nice job! Since this is a FAQ page for a variety of tools, perhaps we should move the [Queries FAQ](https://wormhole.com/docs/build/applications/queries/faqs/) content to this page - wdyt?

martin0995 commented:
I added the Queries FAQs to this new page and deleted the [Queries FAQ](https://wormhole.com/docs/build/applications/queries/faqs/) content from the Build section.

eshaben commented on learn/faq/index.md:
I think for formatting purposes, this should be wrapped in 3 backticks

eshaben reviewed (commented):
Noticed a few more things after looking at this with fresh eyes 👀 

I think we might need to revisit the placement of this page. Most of these are questions users would run into when developing with Wormhole, as opposed to when they're learning about the basics and such.

eshaben commented on build/applications/connect/faqs.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: FAQs
+description: Common questions and answers about Wormhole Connect.
```
This meta description should be between 120-160 characters

eshaben commented on build/applications/connect/faqs.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: FAQs
```
```suggestion
title: Connect FAQs
```

To differentiate this in search engines, lets add the section name to it

eshaben commented on build/applications/queries/faqs.md:
```diff
@@ -0,0 +1,45 @@
+---
+title: FAQs
```
```suggestion
title: Queries FAQs
```

eshaben commented on build/applications/queries/faqs.md:
```diff
@@ -0,0 +1,45 @@
+---
+title: FAQs
+description: Common questions and answers about Wormhole Queries.
```
120-160 characters

eshaben commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+---
+title: FAQs
```
```suggestion
title: Native Token Transfers FAQs
```

eshaben commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+---
+title: FAQs
+description: Common questions and answers about Wormhole NTT.
```
120-160 chars

eshaben commented on build/applications/connect/faqs.md:
```diff
@@ -0,0 +1,50 @@
+description: Common questions and answers about Wormhole Connect.
+---
+
+## Wormhole Connect FAQs
```
This should be an H1 (`#`). This allows us to be flexible with the `title` above for SEO purposes

eshaben commented on build/applications/connect/faqs.md:
```diff
@@ -0,0 +1,50 @@
+
+## Wormhole Connect FAQs
+
+### What types of assets does Connect support? 
```
Then each of these should be H2 (`##`)

eshaben commented on build/applications/queries/faqs.md:
```diff
@@ -1,13 +1,13 @@
 ---
 
-# FAQs about Queries
+## Wormhole Queries FAQs
```
This should be an H1 (`#`)

eshaben commented on build/applications/queries/faqs.md:
```diff
@@ -1,13 +1,13 @@
+## Wormhole Queries FAQs
 
-## What Libraries Are Available to Handle Queries?
+### What libraries are available to handle queries?
```
These should be H2's (`##`)

ilariae replied:
you put suggestions for h3 :)

eshaben commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+description: Common questions and answers about Wormhole NTT.
+---
+
+## Wormhole NTT FAQs
```
Should be an H1 (`#`)

ilariae replied:
```suggestion
# Wormhole NTT FAQs
```

eshaben commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+## Wormhole NTT FAQs
+
+### Do you have an example of how cross-chain lending can be implemented using Wormhole?
```
These should be H2s `(##)`

ilariae replied:
```suggestion
## Do you have an example of how cross-chain lending can be implemented using Wormhole?
```

eshaben commented on build/faq/index.md:
```diff
@@ -0,0 +1,126 @@
+ - [Solana SPL](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/solana/modules/token_bridge/program/src/api/create_wrapped.rs#L128-L145){target=\_blank}
+ - [Attestation VAA and Token Contract Deployment Logic](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/ethereum/contracts/bridge/Bridge.sol#L385-L431){target=\_blank}
+
+## Queries
```
These are now in a separate file, right?

eshaben commented on build/faq/index.md:
```diff
@@ -0,0 +1,126 @@
+
+These resources will guide you through building and deploying custom relayers tailored to your use case.
+
+## Token Bridge
+
+### Can ownership of wrapped tokens be transferred from the Token Bridge?
+
+No, you cannot transfer ownership of wrapped token contracts from the [Token Bridge](/learn/messaging/token-nft-bridge/){target=\_blank} because the Token Bridge deploys and retains ownership of these contracts and tokens.
+
+ - **On EVM chains** - when you attest a token, the Token Bridge deploys a new ERC-20 contract as a beacon proxy. The upgrade authority for these contracts is the Token Bridge contract itself
+ - **On Solana** - the Token Bridge deploys a new SPL token, where the upgrade authority is a Program Derived Address (PDA) controlled by the Token Bridge
+
+The logic behind deploying these token contracts involves submitting an attestation VAA, which allows the Token Bridge to verify and deploy the wrapped token contract on the destination chain.
+
+Relevant contracts:
+
+ - [Ethereum ERC-20](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/ethereum/contracts/bridge/token/Token.sol){target=\_blank}
+ - [Solana SPL](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/solana/modules/token_bridge/program/src/api/create_wrapped.rs#L128-L145){target=\_blank}
+ - [Attestation VAA and Token Contract Deployment Logic](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/ethereum/contracts/bridge/Bridge.sol#L385-L431){target=\_blank}
```
This should probably be under contract integrations FAQ

eshaben commented on build/faq/index.md:
````diff
@@ -0,0 +1,126 @@
+const wh = await wormhole('Testnet', [solana]);
+```
+
+## Relayer
+
+### How do I start developing a custom relayer?
+
+Previously referred to as specialized relayers, custom relayers allow you to build and tailor relayers to fit your specific use case. To get started, refer to the following resources:
+
+ - [Custom Relayers documentation](/docs/learn/infrastructure/relayer/#custom-relayers){target=\_blank}
+ - [Relayer Engine](/docs/infrastructure/relayers/run-relayer/#get-started-with-the-relayer-engine){target=\_blank}
+ - [Run a Custom Relayer](/docs/infrastructure/relayers/run-relayer/){target=\_blank}
+
+These resources will guide you through building and deploying custom relayers tailored to your use case.
````
This should probably be under contract integrations FAQ

eshaben commented on build/faq/index.md:
````diff
@@ -0,0 +1,126 @@
+
+Following these steps, you can manually submit a VAA in the proper format to a destination chain.
+
+## TypeScript SDK
+
+### Why does the `toNative` function in the TypeScript SDK return an error?
+
+The `toNative` function may return an error if the platform-specific module (such as Solana or EVM) is not correctly imported or passed into the Wormhole constructor.
+
+To fix this, ensure the relevant platform module is imported and included when initializing Wormhole. For example, if you're working with Solana, make sure to import the Solana module and pass it into the Wormhole constructor like this:
+
+```typescript
+import solana from '@wormhole-foundation/sdk/solana';
+const wh = await wormhole('Testnet', [solana]);
+```
````
I'd probably put this in the TypeScript SDK page for now. We can add more FAQ once we convert that page into a section

eshaben commented on build/faq/index.md:
```diff
@@ -0,0 +1,126 @@
+
+# FAQs
+
+## Wormholescan API
```
Toolkit section?

eshaben commented on build/faq/index.md:
````diff
@@ -0,0 +1,126 @@
+    curl -X GET "https://api.wormholescan.io/api/v1/operations?address=0x05c009C4C1F1983d4B915C145F4E782de23d3A38" -H "accept: application/json"
+    ```
+
+### How can I manually submit a VAA to a destination chain in the correct format?
````
Why would someone want to do this?

martin0995 replied:
This was a common question obtained from the Discord FAQs

eshaben reviewed (commented):
I'm not sold on having an FAQ directly under Build. I think the reason behind splitting the FAQ up into specific directories is that so they're easier to find and are more fruitful for users focusing on a specific topic. It seems like having it directly under Build might not be the best solution, unless the questions are generic. Otherwise, we should consider why the user might be asking these questions and try to put the FAQ pages near related content or even as sections within specific content, if that makes more sense. 

If you think the questions are generic enough to be put under build, then you'll have to get an icon for FAQ for the dropdown menus from the designers and we'll have to publish this PR and the mkdocs PR at the same time.

Requesting @ilariae and @evgeniko reviews on this for additional opinions

ilariae commented on build/applications/connect/faqs.md:
```diff
@@ -0,0 +1,50 @@
+
+### Which functions or events does Connect rely on for NTT integration? 
+
+Connect relies on the NTT SDK  for integration, with platform-specific implementations for Solana and EVM. The critical methods involved include initiate and redeem functions and rate capacity methods. These functions ensure Connect can handle token transfers and manage chain-rate limits.
```
```suggestion
Connect relies on the NTT SDK for integration, with platform-specific implementations for Solana and EVM. The critical methods involved include initiate and redeem functions and rate capacity methods. These functions ensure Connect can handle token transfers and manage chain-rate limits.
```

ilariae commented on build/applications/queries/faqs.md:
```suggestion
## Are there any query examples?
```

ilariae commented on build/applications/queries/faqs.md:
```diff
@@ -27,18 +27,18 @@ Certainly. You can find a complete guide on the [Use Queries page](/docs/build/a
 - [Solana Queries Verification Example](https://github.com/wormholelabs-xyz/example-queries-solana-verify){target=\_blank}
 
-## What Is the Format of the Response Signature?
+### What is the format of the response signature?
```
```suggestion
## What is the format of the response signature?
```

ilariae commented on build/applications/queries/faqs.md:
```diff
@@ -27,18 +27,18 @@ Certainly. You can find a complete guide on the [Use Queries page](/docs/build/a
     If you are used to `ecrecover` you will notice that the `v` byte is `0` or `1` as opposed to `27` or `28`. The `signaturesToEvmStruct` method in the [Query TypeScript SDK](https://npmjs.com/package/@wormhole-foundation/wormhole-query-sdk){target=\_blank} accounts for this as well as structuring the signatures into an `IWormhole.SignatureStruct[]`.
 
-## Can Anyone Run a Query Proxy Server?
+### Can anyone run a query proxy server?
```
```suggestion
## Can anyone run a query proxy server?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+Alternatively, you can also implement cross-chain lending using [Wormhole’s core messaging](/learn/messaging/native-token-transfers/){target=\_blank} instead of the Token Bridge, which avoids the limitations imposed by governor limits. ETH would be custodied on Ethereum, and BNB on the Binance spoke during this setup. When a user deposits ETH on Ethereum, a core bridge message is sent to the hub  ...
+
+### What causes the "No protocols registered for Evm" error in Wormhole SDK?
```
```suggestion
## What causes the "No protocols registered for Evm" error in Wormhole SDK?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
````diff
@@ -0,0 +1,88 @@
+To resolve this issue, ensure you have imported the appropriate Wormhole SDK package for EVM environments. The necessary package for handling NTT on EVM chains is `@wormhole-foundation/sdk-evm-ntt`. Here's the correct import statement:
+
+```rust
+import '@wormhole-foundation/sdk-evm-ntt';
````
should this be between ' ' ?

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+By importing this package, the Wormhole SDK can register and utilize the required protocols for EVM chains, enabling cross-chain token transfers using the NTT framework. Ensure to include this import at the start of your code, especially before attempting any interactions with EVM chains in your project.
+
+### How can I transfer ownership of NTT to a multisig?
```
```suggestion
## How can I transfer ownership of NTT to a multisig?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+2. **Claim ownership** - the multisig must then claim ownership via the `claim_ownership` instruction. If not claimed, the current owner can cancel the transfer
+3. **Single-step transfer (Riskier)** - you can also use the `transfer_ownership_one_step_unchecked` method to transfer ownership in a single step, but if the new owner cannot sign, the contract may become locked. Be cautious and ensure the new owner is a Program Derived Address (PDA)
+
+
```
```suggestion
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+3. **Single-step transfer (Riskier)** - you can also use the `transfer_ownership_one_step_unchecked` method to transfer ownership in a single step, but if the new owner cannot sign, the contract may become locked. Be cautious and ensure the new owner is a Program Derived Address (PDA)
+
+
+### How can I specify a custom RPC for NTT?
```
```suggestion
## How can I specify a custom RPC for NTT?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+In most cases, the transaction will resume automatically once the rate limits are adjusted, and the UI will guide you through the redemption process.
+
+### What are the challenges of deploying NTT to non-EVM chains?
```
```suggestion
## What are the challenges of deploying NTT to non-EVM chains?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+NTT requires the same transceiver for all routes, limiting flexibility when deploying across EVM and non-EVM chains. For example, if you're deploying to Ethereum, Arbitrum, and Solana, you can't use Wormhole and Axelar as transceivers because Axelar doesn't support Solana. This constraint forces integrators to use a single transceiver (e.g., Wormhole) for all chains, reducing flexibility in optim ...
+
+### Does the NTT manager function as an escrow account for a hub chain?
```
```suggestion
## Does the NTT manager function as an escrow account for a hub chain?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+Yes, the NTT manager acts like an escrow account for non-transferable tokens on a hub chain. To manage non-transferable tokens, you would add the NTT manager to the allowlist, ensuring that only the NTT manager can hold and control the tokens as they are transferred across chains.
+
+### Which functions or events does Connect rely on for NTT integration?
```
```suggestion
## Which functions or events does Connect rely on for NTT integration?
```

ilariae commented on build/contract-integrations/native-token-transfers/faqs.md:
```diff
@@ -0,0 +1,88 @@
+
+These functions ensure Connect can handle token transfers and manage chain-rate limits.
+
+### How does the relayer contract determine which transceiver to call?
```
```suggestion
## How does the relayer contract determine which transceiver to call?
```

eshaben commented on build/index.md:
```diff
@@ -52,4 +52,12 @@ Whether you’re developing front-end applications using Wormhole's application-
 
     [:octicons-arrow-right-16: Explore reference](/docs/build/reference/)
 
+-   :octicons-question-16:{ .lg .middle } **FAQs**
```
This is not necessary here anymore since there isn't an FAQ page directly under Build. But looks like the connect, toolkit, contract integrations, and native token transfers index pages still need to be updated

eshaben commented on build/applications/queries/index.md:
```diff
@@ -27,12 +27,4 @@ Wormhole Queries offers on-demand access to Guardian-attested on-chain data via
 
     [:octicons-arrow-right-16: Get hands-on](/docs/build/applications/queries/use-queries/)
 
--   :octicons-book-16:{ .lg .middle } **Query FAQs**
```
This should be added back, sorry! 😬

eshaben reviewed (commented):
Nice! Just need to update the index pages and this should be good to go!

eshaben commented on build/contract-integrations/index.md:
```diff
@@ -59,4 +59,12 @@ The content in this section will teach you how to create smart contracts that in
 
     [:octicons-arrow-right-16: Set up a dev environment](/docs/build/contract-integrations/dev-env/)
 
+-   :octicons-book-16:{ .lg .middle } **Contract Integrations FAQs**
```
```suggestion
-   :octicons-question-16:{ .lg .middle } **Contract Integrations FAQs**
```

This is a nitpick, but lets use the question mark for FAQs

eshaben commented on build/contract-integrations/native-token-transfers/index.md:
```diff
@@ -27,4 +27,12 @@ This section provides comprehensive guidance on configuring, deploying, and mana
 
     [:octicons-arrow-right-16: Configure your NTT deployment](/docs/build/contract-integrations/native-token-transfers/configuration/)
 
+-   :octicons-book-16:{ .lg .middle } **NTT FAQs**
```
```suggestion
-   :octicons-question-16:{ .lg .middle } **NTT FAQs**
```

eshaben reviewed (commented):
Sorry just one tiny nit 😬  but looks good! ✨

eshaben reviewed (approved):
Beautiful! 🚀
